### PR TITLE
Add example for `Camera::viewport_to_world`

### LINKED ISSUE #7177: Demonstrate how to use `Camera::viewport_to_word`
## What problem does this solve or what need does it fill?

Seems `Camera::viewport_to_word` is similiar to [bevy_mode_raycasting](https://github.com/aevyrie/bevy_mod_raycast). I prefer engine built-in functionality over third-party plugins. But there is no any example to demonstrate how to use it.

## What solution would you like?

Create an example to demonstrate how to use `Camera::viewport_to_word`.

### PR DESCRIPTION
**Pull request by devil-ira**
Fixes #7177

### CODE REVIEW

**Review by mockersf — CHANGES_REQUESTED** [review]
could you add an example description at the tope of the example in comments?

File: examples/3d/3d_viewport_to_world.rs [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    }
+
+    let Some(cursor_position) = windows.single().cursor_position() else { return; };
+
+    let Some(ray) = camera.viewport_to_world(camera_transform, cursor_position) else { return; };
+
+    let Some(distance) = ray.intersect_plane(Vec3::ZERO, Vec3::Y) else { return; };
+
+    let point = ray.get_point(distance);
+    transform.translation = point + Vec3::Y * 0.5;
```
Comment (rparrett) [inline examples/3d/3d_viewport_to_world.rs:33]: I'm wondering if this section might be somewhat magical for new users.

- `Vec3::ZERO, Vec3::Y` duplicates the position / normal of the `Plane` we built in `setup`. 
- `Vec3::Y * 0.5` duplicates the offset we defined for the cube in `setup`.

I played around with this a smidge, and I think

`intersect_plane(ground.translation, ground.up())` reads well.

Querying for `Aabb` and placing the cube based on its midpoint feels sort of messy. Maybe a `CUBE_HEIGHT` constant or something would be clearer for the example?

Or possibly just some comments would help.

File: examples/README.md [hunk 2]
```diff
@@ -111,6 +111,7 @@ Example | Description
 [3D Gizmos](../examples/3d/3d_gizmos.rs) | A scene showcasing 3D gizmos
 [3D Scene](../examples/3d/3d_scene.rs) | Simple 3D scene with basic shapes and lighting
 [3D Shapes](../examples/3d/3d_shapes.rs) | A scene showcasing the built-in 3D shapes
+[3D Viewport To World](../examples/3d/3d_viewport_to_world.rs) | Use the cursor to move a cube through the scene
```
Comment (rparrett) [inline examples/README.md:114]: This description is no longer accurate.

**Review by rparrett — APPROVED** [review]

**Comment by devil-ira** [comment]
I've added an example for 2D as well.

File: examples/3d/3d_viewport_to_world.rs [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    mut meshes: ResMut<Assets<Mesh>>,
+    mut materials: ResMut<Assets<StandardMaterial>>,
+) {
+    // plane
+    commands.spawn((
+        PbrBundle {
+            mesh: meshes.add(Mesh::from(shape::Plane {
+                size: 20.0,
+                subdivisions: 0,
+            })),
```
Comment (rparrett) [inline examples/3d/3d_viewport_to_world.rs:55]:
```suggestion
            mesh: meshes.add(Mesh::from(shape::Plane::from_size(20.))),
```

or

```rust
            mesh: meshes.add(Mesh::from(shape::Plane {
                size: 20.0,
                ..default()
            })),
```

**Review by rparrett — APPROVED** [review]

**Review by mockersf — APPROVED** [review]

**Pull request merged**